homeassistant/components/unifiprotect/views.py

```diff
@@ -0,0 +1,211 @@
+    if code == HTTPStatus.BAD_REQUEST:
+        return web.Response(body=message, status=code)
+    return web.Response(status=code)
+
+
+@callback
+def _400(message: Any) -> web.Response:
+    return _client_error(message, HTTPStatus.BAD_REQUEST)
+
+
+@callback
+def _403(message: Any) -> web.Response:
+    return _client_error(message, HTTPStatus.FORBIDDEN)
+
+
+@callback
+def _404(message: Any) -> web.Response:
+    return _client_error(message, HTTPStatus.NOT_FOUND)
+
+
+@callback
+def _validate_event(event: Event) -> None:
+    if event.camera is None:
+        raise ValueError("Event does not have a camera")
+    if not event.camera.can_read_media(event.api.bootstrap.auth_user):
+        raise PermissionError(f"User cannot read media from camera: {event.camera.id}")
+
+
+class ProtectProxyView(HomeAssistantView):
+    """Base class to proxy request to UniFi Protect console."""
+
+    requires_auth = True
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize a thumbnail proxy view."""
+        self.hass = hass
+        self.data = hass.data[DOMAIN]
+
+    def _get_data_or_404(self, nvr_id: str) -> ProtectData | web.Response:
+        all_data: list[ProtectData] = []
+
+        for data in self.data.values():
+            if isinstance(data, ProtectData):
+                if data.api.bootstrap.nvr.id == nvr_id:
+                    return data
+                all_data.append(data)
+        return _404("Invalid NVR ID")
+
+
+class ThumbnailProxyView(ProtectProxyView):
+    """View to proxy event thumbnails from UniFi Protect."""
+
+    url = "/api/ufp/thumbnail/{nvr_id}/{event_id}"
+    name = "api:ufp_thumbnail"
+
+    async def get(
+        self, request: web.Request, nvr_id: str, event_id: str
+    ) -> web.Response:
+        """Get Event Thumbnail."""
+
+        data = self._get_data_or_404(nvr_id)
+        if isinstance(data, web.Response):
+            return data
+
+        width: int | str | None = request.query.get("width")
+        height: int | str | None = request.query.get("height")
+
+        if width is not None:
+            try:
+                width = int(width)
+            except ValueError:
+                return _400("Invalid width param")
+        if height is not None:
+            try:
+                height = int(height)
+            except ValueError:
+                return _400("Invalid height param")
+
+        try:
+            thumbnail = await data.api.get_event_thumbnail(
+                event_id, width=width, height=height
+            )
+        except ClientError as err:
+            return _404(err)
+
+        if thumbnail is None:
+            return _404("Event thumbnail not found")
+
+        return web.Response(body=thumbnail, content_type="image/jpeg")
+
+
+class VideoProxyView(ProtectProxyView):
+    """View to proxy video clips from UniFi Protect."""
+
+    url = "/api/ufp/video/{nvr_id}/{camera_id}/{start}/{end}"
```

Comment:
```suggestion
    url = "/api/unifiprotect/video/{nvr_id}/{camera_id}/{start}/{end}"
```

Please match the namespace of the integration